src/index.ts

```diff
@@ -125,17 +135,9 @@ export class CapBase<T> {
       });
     }
 
-    const router = "router" in args ? args.router : await CapRouter.init({ host: args.routerHost, canisterId: args.routerCanisterId });
-
-    const { canister } = await router.get_token_contract_root_bucket({
-      tokenId: Principal.fromText(args.tokenId),
-    });
-
-    if (!canister?.[0]) throw Error(`Token ${args.tokenId} not in cap`);
-
     return Actor.createActor(idlFactory, {
       agent,
-      canisterId: canister[0],
+      canisterId: args.tokenId,
```

Comment:
It not gonna work. Here you cannot provide tokenId directly you have to pass the rootTokenId in other case u gonna create an actor from token

Reply:
You might do it here https://github.com/Psychedelic/cap-js/pull/6/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R149-R153 before call `createActor`, so you could make  createActor sync but you should keep `initialize` async. If it is not possible you should revert this PR: https://github.com/Psychedelic/cap-js/pull/9/files

Reply:
6e4b7d1 I accommodated both scenarios